tests/lts_compatibility.py

```diff
@@ -259,65 +267,74 @@ def run_ledger_compatibility_since_first(args, use_snapshot):
+                    "library_dir": library_dir,
+                    "txs": txs,
+                    "jwt_issuer": jwt_issuer,
+                    "version": major_version,
+                }
+                if idx == 0:
+                    LOG.info(f"Starting new service (version: {version})")
+                    network = infra.network.Network(**network_args)
+                    network.start_and_join(args)
+                else:
+                    LOG.info(f"Recovering service (new version: {version})")
+                    network = infra.network.Network(
+                        **network_args, existing_network=network
+                    )
+                    network.start_in_recovery(
+                        args,
+                        ledger_dir,
+                        committed_ledger_dir,
+                        snapshot_dir=snapshot_dir,
+                    )
+                    network.recover(args)
+
+                nodes = network.get_joined_nodes()
+                primary, _ = network.find_primary()
+
+                # Verify that all nodes run the expected CCF version
+                for node in nodes:
+                    # Note: /node/version endpoint was added in 2.x
+                    if not node.version or node.version > 1:
+                        with node.client() as c:
+                            r = c.get("/node/version")
+                            expected_version = node.version or args.ccf_version
+                            version = r.body.json()["ccf_version"]
+                            assert (
+                                r.body.json()["ccf_version"] == expected_version
+                            ), f"Node version is not {expected_version}"
+
+                # Rollover JWKS so that new primary must read historical CA bundle table
+                # and retrieve new keys via auto refresh
+                jwt_issuer.refresh_keys()
+                jwt_issuer.wait_for_refresh(network)
+
+                issue_activity_on_live_service(network, args)
+
+                snapshot_dir = (
+                    network.get_committed_snapshots(primary) if use_snapshot else None
+                )
+                ledger_dir, committed_ledger_dir = primary.get_ledger(
+                    include_read_only_dirs=True
+                )
+                network.stop_all_nodes(verbose_verification=False)
 
-            # Check that the ledger can be parsed
-            ccf.ledger.Ledger([committed_ledger_dir]).get_latest_public_state()
+                # Check that ledger and snapshots can be parsed
+                ccf.ledger.Ledger([committed_ledger_dir]).get_latest_public_state()
+                if snapshot_dir:
+                    for s in os.listdir(snapshot_dir):
+                        with ccf.ledger.Snapshot(
+                            os.path.join(snapshot_dir, s)
+                        ) as snapshot:
+                            snapshot.get_public_domain()
```

Comment:
```suggestion
```
This will fail because `1.x`'s Python doesn't know about `Snapshot`.

Reply:
Ok, so we need to cherry-pick #2644?

Reply:
Either that or simply skip this check here as this is test-only?